src/Libraries/Microsoft.Extensions.Caching.Hybrid/Internal/DefaultHybridCache.L2.cs

```diff
@@ -95,13 +95,26 @@ internal void SetL1<T>(string key, CacheItem<T> value, HybridCacheEntryOptions?
         if (value.TryReserve())
         {
             // based on CacheExtensions.Set<TItem>, but with post-eviction recycling
-            using var cacheEntry = _localCache.CreateEntry(key);
+
+            // intentionally use manual Dispose rather than "using"; confusingly, it is Dispose()
+            // that actually commits the add - so: if we fault, we don't want to try
+            // committing a partially configured cache entry
+            var cacheEntry = _localCache.CreateEntry(key);
```

Comment:
As per our coding style guide, we don't use `var` unless the type is obvious.
https://github.com/dotnet/extensions/blob/main/CONTRIBUTING.md#dos-and-donts

Reply:
addressed in [5e3bdba](https://github.com/dotnet/extensions/pull/5420/commits/5e3bdbafb8f2b59c811c7d66fb225a11ef379e2a)